tools/frr-reload.py

```diff
@@ -757,7 +757,7 @@ def ignore_delete_re_add_lines(lines_to_add, lines_to_del):
                     bfd_nbr = "neighbor %s" % nbr
 
                     for (ctx_keys, add_line) in lines_to_add:
-                        re_add_nbr_bfd_timers = re.search(r'neighbor (\S+) bfd (\S+) (\S+) (\S+)', add_line)
+                        re_add_nbr_bfd_timers = re.search(r'neighbor bfd_nbr bfd (\S+) (\S+) (\S+)', add_line)
```

Comment:
just quick question: how this gonna work? I just want to understand, nothing special :) isn't this derived from above (`bfd_nbr = "neighbor %s" % nbr`)?

Reply:
What you've stated is actually why it now works.  This chunk of frr-reload.py is looking for lines in that are in both the "add" and "delete" list in order to know whether a direct replacement can be done instead.  This chunk of code was gathering the exact neighbor the bfd change was being made on and then searching in the "add" list for _any_ neighbor with a bfd timer change.  If there were multiple bfd neighbor timer changes, it would find a match in the add list twice (the one that matches the exact neighbor and also the other one(s)).  By changing the search, we only find and act on the one that exactly matches.

Reply:
makes sense, just the code looks strange :)